### PR TITLE
[WIP] Spring Cloud AWS w/ AWS SDK v2

### PR DESCRIPTION
**Pull request by driverpt**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
1-1 Port for Spring Cloud AWS messaging

## :bulb: Motivation and Context
#20 

## :green_heart: How did you test it?
Still refactoring tests and adding integration tests

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: v3/spring-cloud-aws-v3-context/src/main/java/io/awspring/v3/cloud/context/annotation/OnAwsCloudEnvironmentCondition.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.ConfigurationCondition;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * @author Agim Emruli
+ * @author Gleb Schukin
+ * @author Bernardo Martins
+ */
+@Deprecated
```
Comment (driverpt) [inline v3/spring-cloud-aws-v3-context/src/main/java/io/awspring/v3/cloud/context/annotation/OnAwsCloudEnvironmentCondition.java:30]: Should we delete all @Deprecated stuff? After all, it's a Major release

Comment (driverpt) [reply v3/spring-cloud-aws-v3-context/src/main/java/io/awspring/v3/cloud/context/annotation/OnAwsCloudEnvironmentCondition.java]: @maciejwalkowiak

File: v3/spring-cloud-aws-v3-integration-test/src/test/java/io/awspring/cloud/v3/it/LocalStackIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package io.awspring.cloud.v3.it;
```
Comment (driverpt) [inline v3/spring-cloud-aws-v3-integration-test/src/test/java/io/awspring/cloud/v3/it/LocalStackIntegrationTest.java:1]: @maciejwalkowiak , can you please review this one?

File: v3/spring-cloud-aws-v3-integration-test/src/test/java/io/awspring/cloud/v3/it/IntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package io.awspring.cloud.v3.it;
```
Comment (driverpt) [inline v3/spring-cloud-aws-v3-integration-test/src/test/java/io/awspring/cloud/v3/it/IntegrationTest.java:1]: @maciejwalkowiak , can you please review this one?

**Comment by maciejwalkowiak** [comment]
@driverpt the plan is not only to migrate to AWS SDK v2 but also take this chance to revisit all the concepts and integrations we have. Spring Cloud AWS was created in 2013, AWS has changed a lot since then and some integrations or some patterns do not make sense any longer. 

I feel really bad for closing this PR as you have put a lot of work into it, but we can't and don't want to migrate just like that.

**Comment by driverpt** [comment]
I think we should start simple. 1-1 Port and go from there. It's easier

**Comment by driverpt** [comment]
@maciejwalkowiak , am i going to be left out here? Seems to me that you're not willing to accept any contributions.

Is this the case?

**Comment by maciejwalkowiak** [comment]
Of course we do accept contributions (see commit history) but 3.x is not yet in the state when we can easily accept contributions. Keep in mind that submitting PR is one thing, we are later responsible for maintaining anything that got merged.

**Comment by driverpt** [comment]
Alright, but can we be included in the development? I still think we should first migrate what we currently have as-is refactored for the AWS SDK v2 clients, and then improve it

**Comment by maciejwalkowiak** [comment]
We have just closed 2.3.4 and 2.4.0. Now we switch focus to 3.x again and come up with guidelines for contributors.

**Comment by driverpt** [comment]
It would be nice if you actually accepted contributions, instead of turning down every single PR regarding 3.0.x with no discussion whatsoever and there is plenty of people happy to contribute with ideas and new concepts

**Comment by maciejwalkowiak** [comment]
Once we bring 3.x to the state where we can accept contributions we will be more than happy to accept contributions.

**Pull request closed without merging**